openedx/features/edly/tests/test_utils.py

```diff
@@ -346,3 +348,56 @@ def test_create_learner_link_with_permission_groups(self):
 
         edly_user = create_learner_link_with_permission_groups(edly_user)
         assert edly_user.groups.filter(name=edly_user_group).exists()
+
+    def test_get_current_site_invalid_certificate_context_data_without_site_configuration(self):
+        """
+        Test method returns correct data without site configuration.
+        """
+        test_default_certificate_html_configurations = {
+            'default': {
+                'accomplishment_class_append': 'accomplishment-certificate',
+                'platform_name': 'fake-platform-name',
+                'logo_src': 'fake-logo-path',
+                'logo_url': 'fake-logo-url',
+                'company_verified_certificate_url': 'fake-verified-certificate-url',
+                'company_privacy_url': 'fake-privacy-url',
+                'company_tos_url': 'fake-tos-url',
+                'company_about_url': 'fake-about-url'
+            }
+        }
+
+        expected_current_site_context_data = test_default_certificate_html_configurations['default']
+        curent_site_context_data = get_current_site_invalid_certificate_context_data(test_default_certificate_html_configurations)
+        assert expected_current_site_context_data == curent_site_context_data
+
+    def test_get_current_site_invalid_certificate_context_data_with_site_configuration(self):
+        """
+        Test method returns correct data with site configuration.
+        """
+        test_default_certificate_html_configurations = {
+            'default': {
+                'accomplishment_class_append': 'accomplishment-certificate',
+                'platform_name': 'fake-platform-name',
+                'logo_src': 'fake-logo-path',
+                'logo_url': 'fake-logo-url',
+                'company_verified_certificate_url': 'fake-verified-certificate-url',
+                'company_privacy_url': 'fake-privacy-url',
+                'company_tos_url': 'fake-tos-url',
+                'company_about_url': 'fake-about-url'
+            }
+        }
+        site_configuration = SiteConfigurationFactory(site=self.request.site)
+        site_configuration.values.update({
```

Comment:
are there some default values that the factory generates? why do we need to update it separately? can it be done during factory creation?

Reply:
Done